### PR TITLE
Make site generation more generic

### PR DESCRIPTION
**Pull request by ericsagnes**
Wow! I love the idea of a static site generator in nix :)

I hope this can help to make this tool more generic.

Please review carefully, as this is my first contribution to this project and I am not very familiar on how to test that all is working fine yet.

Major changes:
- ~~separated the templates in a dedicated nix file~~
- added a configuration file

update:
- moved lib functions in a separate file
- moved site generation in lib
- added preview mode
- added page generation function to lib
- moved each templates in a dedicated files

### CODE REVIEW

**Comment by ericsagnes** [comment]
cc @garbas

**Comment by ericsagnes** [comment]
Hacking on this repo is highly addictive :)
Ended in touching much more code than planned, so I hope I didn't took this in a wrong direction.

**Comment by Mic92** [comment]
Looks definitely cleaner

**Comment by ericsagnes** [comment]
By the way, please let me know if this PR commits should be squashed or not.
For now I let all the commits as they seem granular enough, but I can squash everything if it is a better fit to this repository policy.

File: default.nix [hunk 1]
```diff
@@ -1,465 +1,60 @@
...
-                    </li>
-                    -->
-                  </ul>
-                </div>
-              </div>
-            '';
-        };
+*/
+{ pkgs ? import <nixpkgs> {}
+, current_timestamp  # date -u +%Y-%m-%dT%TZ
```
Comment (zimbatm) [inline default.nix]: this should be `currentTimestamp` to follow the nix variable naming conventions

**Review by zimbatm — APPROVED** [review]
LGTM with small nitpick

File: default.nix (hunk 1 above)
Comment (ericsagnes) [reply default.nix]: Fixed, thanks!

**Comment by garbas** [comment]
tnx.

**Pull request merged**